### ISSUE TITLE
Lot's of errors shown of Performance Reporting on Android

### ISSUE THREAD

**Issue by wagenheimer** [Demigiant/dotween#152]
Hi!

I'm using Download DOTween v1.1.640. Unity 2017.1.1. I'm not sure, but it seems to happens only on Android Devices.

I have tons of bug reports with this error :

IndexOutOfRangeException: Array index is out of range.
```
DG.Tweening.Core.TweenManager.RemoveActiveTween (DG.Tweening.Tween t)
DG.Tweening.Core.TweenManager.Despawn (DG.Tweening.Tween t, Boolean modifyActiveLists)
DG.Tweening.Core.TweenManager.DespawnActiveTweens (System.Collections.Generic.List`1 tweens)
DG.Tweening.Core.TweenManager.Update (UpdateType updateType, Single deltaTime, Single independentTime)
DG.Tweening.Core.DOTweenComponent.Update ()
```

IndexOutOfRangeException: Array index is out of range.
```
DG.Tweening.Core.TweenManager.ReorganizeActiveTweens ()
DG.Tweening.Core.TweenManager.Update (UpdateType updateType, Single deltaTime, Single independentTime)
DG.Tweening.Core.DOTweenComponent.Update ()
```

IndexOutOfRangeException: Array index is out of range.
```
DG.Tweening.Core.TweenManager.ReorganizeActiveTweens ()
DG.Tweening.Core.TweenManager.AddActiveTween (DG.Tweening.Tween t)
DG.Tweening.Core.TweenManager.GetTweener[T1,T2,TPlugOptions] ()
DG.Tweening.DOTween.ApplyTo[T1,T2,TPlugOptions] (DG.Tweening.Core.DOGetter`1 getter, DG.Tweening.Core.DOSetter`1 setter, .T2 endValue, Single duration, DG.Tweening.Plugins.Core.ABSTweenPlugin`3 plugin)
DG.Tweening.DOTween.To (DG.Tweening.Core.DOGetter`1 getter, DG.Tweening.Core.DOSetter`1 setter, Vector3 endValue, Single duration)
DG.Tweening.ShortcutExtensions.DOScale (UnityEngine.Transform target, Vector3 endValue, Single duration)

```

IndexOutOfRangeException: Array index is out of range.
```
DG.Tweening.Core.TweenManager.ReorganizeActiveTweens ()
DG.Tweening.Core.TweenManager.AddActiveTween (DG.Tweening.Tween t)
DG.Tweening.Core.TweenManager.GetTweener[T1,T2,TPlugOptions] ()
DG.Tweening.DOTween.ApplyTo[T1,T2,TPlugOptions] (DG.Tweening.Core.DOGetter`1 getter, DG.Tweening.Core.DOSetter`1 setter, .T2 endValue, Single duration, DG.Tweening.Plugins.Core.ABSTweenPlugin`3 plugin)
DG.Tweening.DOTween.ToArray (DG.Tweening.Core.DOGetter`1 getter, DG.Tweening.Core.DOSetter`1 setter, UnityEngine.Vector3[] endValues, System.Single[] durations)
DG.Tweening.DOTween.Shake (DG.Tweening.Core.DOGetter`1 getter, DG.Tweening.Core.DOSetter`1 setter, Single duration, Vector3 strength, Int32 vibrato, Single randomness, Boolean ignoreZAxis, Boolean vectorBased, Boolean fadeOut)
DG.Tweening.DOTween.Shake (DG.Tweening.Core.DOGetter`1 getter, DG.Tweening.Core.DOSetter`1 setter, Single duration, Vector3 strength, Int32 vibrato, Single randomness, Boolean fadeOut)
DG.Tweening.ShortcutExtensions.DOShakeScale (UnityEngine.Transform target, Single duration, Vector3 strength, Int32 vibrato, Single randomness, Boolean fadeOut)
```

Is there something I can look to fix this?

**Comment by wagenheimer** [comment]
Please, is there anything I can do. I love DOTween and I bought DOTweenPRO, but I will have to change to anything else because of this problem.

**Comment by Demigiant** [comment]
Hi Cezar,

I'm sorry, I somehow missed your first post, apologies!

Uhm this seems like a very old problem which was fixed a long time ago. I would really need to see a barebone project that reproduces this. I'm pretty sure it can't be related to the platform, but maybe you're doing something differently in how the tweens happen.

I reckon it might be hard to reproduce, but can you tell me the steps of what you're doing (both in coding, how you create tweens etc, and in action, how you interact with the elements), and maybe that'll help.

**Comment by wagenheimer** [comment]
I do have some Crash Dumps for Windows from my Beta Testers, strangely my game is crashing. It seems related do Change Scale, but I could not reproduce it so I cannot they for sure what is causing it, but it happens when DOTween is scaling it. Here is a call stack and the dump file and the pdb file for debugging. I reported it to Unity.

[error.log](https://github.com/Demigiant/dotween/files/1330582/error.log)
[player_win_x86.pdb.zip](https://github.com/Demigiant/dotween/files/1330584/player_win_x86.zip)
[crash.dmp.zip](https://github.com/Demigiant/dotween/files/1330592/crash.dmp.zip)


Secrets of Magic 2 - Witches and Wizards.exe!__VEC_memcpy() Unknown Non-user code. Symbols loaded.
> Secrets of Magic 2 - Witches and Wizards.exe!BoxCollider::ScaleChanged(void) Unknown Non-user code. Symbols loaded.
Secrets of Magic 2 - Witches and Wizards.exe!Collider::Raycast(class Ray const &,float,struct RaycastHit &) Unknown Non-user code. Symbols loaded.
Secrets of Magic 2 - Witches and Wizards.exe!GameObject::SendMessageAny(class MessageIdentifier const &,struct MessageData &) Unknown Non-user code. Symbols loaded.
Secrets of Magic 2 - Witches and Wizards.exe!InitializeRectTransformAnimationBindingInterface(void) Unknown Non-user code. Symbols loaded.
Secrets of Magic 2 - Witches and Wizards.exe!Transform::SetLocalScale(class Vector3f const &) Unknown Non-user code. Symbols loaded.
Secrets of Magic 2 - Witches and Wizards.exe!Transform_CUSTOM_INTERNAL_set_localScale(struct MonoObject *,class Vector3f const &) Unknown Non-user code. Symbols loaded.

**Comment by Whyser** [comment]
We are using DOTween 1.1.340 in Unity 5.5.0f3, in which version was this previously fixed? @Demigiant 

We have the same issue, in our live product, with over 200+ errors each day. We can't reproduce the issue either. It's on both Android and iOS.
I was about to post some stacktraces, but they are the same as wagenheimer (maybe even less details).
We think it has something to do with the following (we've seen some faint logs tracing back to these):

```
badgeNumbericon.transform.DOKill(true);
badgeNumbericon.transform.parent.DOKill(true);
badgeNumbericon.transform.DOPunchScale(Vector2.one * 1.5f, 0.5f);
badgeNumbericon.transform.parent.DOPunchScale(Vector2.one * 0.05f, 0.5f, 7, 0.7f);
```
and
```
Gems[i].DOKill(true);
wheelTransform.DOKill(true);
wheelTransform.DORotate(new Vector3(0, 0, degrees + 2*360), spinTime, RotateMode.FastBeyond360).SetEase(Ease.OutCubic)
Gems[i].DOScale(1, 0.2f).SetDelay(0.4f);
Gems[i].DOAnchorPos(wheelTransform.GetComponent<RectTransform>().anchoredPosition,0.4f).SetEase(Ease.InBack).SetDelay(0.2f)
```

In other words, it might have to do with doing DOKill(true) and then tweening again in the same frame. But I don't see if it's possible since your code is synchronous(?). 

Sorry that we can't give you more info.

Btw, is there any way to log properly to know which tween actually causes the issue? Often when DOTween gives error, we have no clue what causes it other then, maybe what kind of tween it was (for example set__scale)

Thanks for your great work!

**Comment by Demigiant** [comment]
@Whyser The kill issue indeed caused similar errors, but should be related to the fix introduced in v1.1.575, so I'd suggest to update and let me know if it's alright :) About getting more error info, those stack traces are indeed what I need. The problem is that they're very rare (solving the original issue was a true pain) and caused by some peculiar usages, and reproducing them is usually hard.

@wagenheimer About this last error you mentioned, the dump exception is:

> The thread tried to read from or write to a virtual address for which it does not have the appropriate access

so that should definitely not be related to DOTween, maybe to something else happening during that scale.
About your initial issue, are you using DOTween.Kill or DOTween.Clear in some peculiar ways (or at all)? Is recycling on or off? Can you post a screenshot of DOTween's Utility Panel settings, so I can see them all, and also of the DOTween inspector during play mode?

**Comment by Whyser** [comment]
Thank you Demigiant, I will try upgrade and hope that the issues disappear!

**Comment by wagenheimer** [comment]
Hi, I checked here : 

- I do use some transform.DOKill(false); on OnDisable.
- I use a lot of DOTween.Kill(Object.transform) on several situations;
- And do use some Tween.Kill()

![dotween](https://user-images.githubusercontent.com/315513/30861677-e793078c-a2a1-11e7-8e30-4fdcffdc3a3d.png)

**Comment by Whyser** [comment]
Hello again @Demigiant !
We are sorry to say that the errors still happen (a lot less though, although that might have been us removing some of the cases where it happens).

It seems to be when kalling kill within a kill, something like:

```
tranforms.DOScale(1f, 0.5f).OnComplete(()=>{
    tranform.DOKill(false);
   Destroy(tranform);
})

void OnDestroy()
{
    tranform.DOKill(true);
}
```

If object is destroyed from the outside for some reason, the DOKill(true) will cause the onComplete to be called which in turn will call DOKill(false). This is not the exact case since, one would normally not Destroy something that is null (writing on top of my head here), but I think you get the point.

**Comment by Demigiant** [comment]
Agh!!! Damn, I'm gonna check that asap, hopefully today.

**Comment by Whyser** [comment]
Thanks man! Here's a real log (at work now):

`IndexOutOfRangeException: Array index is out of range. DG.Tweening.Core.TweenManager.RemoveActiveTween (DG.Tweening.Tween t) DG.Tweening.Core.TweenManager.FilteredOperation (OperationType operationType, FilterType filterType, System.Object id, Boolean optionalBool, Single optionalFloat, System.Object optionalObj, System.Object[] optionalArray) DG.Tweening.DOTween.Kill (System.Object targetOrId, Boolean complete) TutorialSliceBase.TweenKiller (Boolean complete) TutorialSliceBase.OnDestroy () TutorialSliceDeepWater.OnDestroy ()`

And in OnDestroy:
```
void OnDestroy()
{
   TweenKiller();
}
```

And inside TweenKiller():
```
 private void TweenKiller(bool complete = true)
    {
        if (infoBox != null)
            infoBox.DOKill(true);
        if (maskCircle != null)
            maskCircle.DOKill(true);
        if (bgImage != null)
            bgImage.DOKill(true);
    }
```

The Tween that "completes":
```
maskCircle.DOScale(Vector2.one * 10f, 0.5f).OnComplete(() => {
                TweenKiller(false);
                if(gameObject != null)
                    Destroy(gameObject);
            });
```

Hope it helps!

**Comment by Demigiant** [comment]
@Whyser  I tried various ways to reproduce this error, including replicating your exact code (both with safe mode disabled and enabled), but everything works correctly here :B Could you send me a sample project that replicates the issue? Maybe I'm missing something.

**Comment by Whyser** [comment]
Unfortunately I can't replicate it either, only see that users are still receiving it. :(

**Comment by Whyser** [comment]
@Demigiant I managed to recreate one of the issues (you should be able to recreate this)!!! See code below:

```
    private void Update()
    {
        if(Input.GetKeyUp(KeyCode.L))
            ReproTweenBug();
    }
    private void ReproTweenBugKiller()
    {
        wheelTransform.DOKill(true);
        Gems[0].DOKill(true);
    }
    private void ReproTweenBug()
    {
        ReproTweenBugKiller();

        var rotation = 2 * 360;
        var duration = rotation / 360f;

        wheelTransform.DORotate(new Vector3(0, 0, rotation), duration, RotateMode.FastBeyond360).SetEase(Ease.OutCubic).OnComplete(() =>
        {
            Gems[0].DOAnchorPos(Vector2.zero, 0.4f).SetEase(Ease.InBack).OnComplete(() =>
            {
                Gems[0].DOKill(true);
            });
        });
    }
```

The "Gems[0]" is a ref to a `RectTransform` the "wheelTransform" is a ref to a `Transform`. To reproduce, press L once, then press L once again before thr DORotate tween finishes (in other words, in 2 seconds). This will cause IndexOutOfRangeException inside the DOTween lib.


The stacktrace if you're interested:

> IndexOutOfRangeException: Array index is out of range.
> DG.Tweening.Core.TweenManager.RemoveActiveTween (DG.Tweening.Tween t) (at D:/DG/_Develop/__UNITY3_CLASSES/_Holoville/__DOTween/_DOTween.Assembly/DOTween/Core/TweenManager.cs:891)
> DG.Tweening.Core.TweenManager.FilteredOperation (OperationType operationType, FilterType filterType, System.Object id, Boolean optionalBool, Single optionalFloat, System.Object optionalObj, System.Object[] optionalArray) (at D:/DG/_Develop/__UNITY3_CLASSES/_Holoville/__DOTween/_DOTween.Assembly/DOTween/Core/TweenManager.cs:552)
> DG.Tweening.DOTween.CompleteAndReturnKilledTot (System.Object targetOrId) (at D:/DG/_Develop/__UNITY3_CLASSES/_Holoville/__DOTween/_DOTween.Assembly/DOTween/DOTween.cs:684)
> DG.Tweening.DOTween.Kill (System.Object targetOrId, Boolean complete) (at D:/DG/_Develop/__UNITY3_CLASSES/_Holoville/__DOTween/_DOTween.Assembly/DOTween/DOTween.cs:744)
> DG.Tweening.ShortcutExtensions.DOKill (UnityEngine.Component target, Boolean complete) (at D:/DG/_Develop/__UNITY3_CLASSES/_Holoville/__DOTween/_DOTween.Assembly/DOTween/ShortcutExtensions.cs:1350)
> WheelBehaviour.ReproTweenBugKiller () (at Assets/Scripts/Shop/WheelBehaviour.cs:358)
> WheelBehaviour.ReproTweenBug () (at Assets/Scripts/Shop/WheelBehaviour.cs:362)
> WheelBehaviour.Update () (at Assets/Scripts/Shop/WheelBehaviour.cs:351)

**Comment by Demigiant** [comment]
I was trying more but never managed, but I just tried your example and it works! Uh, I mean, it doesn't work, but that's what I was looking for. I'm on it! :)

**Comment by Demigiant** [comment]
Fixed! Can you try [v1.1.660](http://dotween.demigiant.com/tmp/DOTween_1_1_660.zip) out and let me know how it goes?

**Comment by Whyser** [comment]
Well, I can't really verify that it works other then the example I gave you. But I will certainly use the updated lib to make sure it doesn't happen to the users.

Thanks for fixing it quickly!

**Comment by Demigiant** [comment]
Let me know if you still see it happen with users. If so, I have another direction to try :)

**Comment by eintopf** [comment]
Hi, we are we seeing this exception too alot, is this maybe a regression?
```
DG.Tweening.Core.TweenManager.RemoveActiveTween (DG.Tweening.Tween t)
DG.Tweening.Core.TweenManager.Despawn (DG.Tweening.Tween t, Boolean modifyActiveLists)
DG.Tweening.Core.TweenManager.DespawnActiveTweens (System.Collections.Generic.List`1 tweens)
DG.Tweening.Core.TweenManager.Update (UpdateType updateType, Single deltaTime, Single independentTime)
DG.Tweening.Core.DOTweenComponent.Update ()
```

Facts: Latest Pro version from AssetStore and Unity 2018.2.13ff.

We are not able to reproduce it yet, but it seems to happen more often in later phase of game. We don't have Tween recycling activated.

It might have something todo with chained OnCompletes. Snippet 1:
`        foreach (var pickup in otherEnergyPickUps)
        {
            if (pickup.isGettingLooted) continue;
            pickup.transform.DOMove(targetPos, 0.25f).SetEase(Ease.OutExpo).OnComplete(() => { pickup.Loot(null); });
        }`

pickup.Loot results in Snippet 2:
`
myTransform.DOScale(Vector3.zero, 0.5f)
        .SetEase(Ease.InOutExpo)
        .OnComplete(() =>
        {
            ...
        });
`

Any help would be appreciated!

**Comment by Demigiant** [comment]
Hi,

Ah, creating tweens inside other tweens' OnComplete shouldn't be a problem, but I'll test it out. I'm writing though to say that I'll be away until the end of the week and I won't be able to test it sooner than next Monday, sorry.

P.S. Is that a warning or a full-on exception? Because there's a safety system I put in place which should capture those errors if they ever happen and throw a warning instead (so I can try to fix that too).

Cheers,
Daniele

**Comment by eintopf** [comment]
Any news on this? I think these are just warnings. We were able to fix it by rolling back our DoTween Pro upgrade for the moment. But it'd be nice to use the newest version. 

So broken in 1.0.075
It works in 1.0.041 (Dotween 1.2.055)

**Comment by Ladimaco** [comment]
I am receiving the following errors with the newest version of DoTween Pro.

This occurs when trying to kill the tween when it completes.  Any idea what I might be doing wrong?  Error appears similar to the other reported in this thread.

Any help is appreciated,
-Larry

`IndexOutOfRangeException: Array index is out of range.
DG.Tweening.Plugins.Core.PathCore.CatmullRomDecoder.GetPoint (Single perc, UnityEngine.Vector3[] wps, DG.Tweening.Plugins.Core.PathCore.Path p, DG.Tweening.Plugins.Core.PathCore.ControlPoint[] controlPoints) (at D:/DG/_Develop/__UNITY3_CLASSES/_Holoville/__DOTween/_DOTween.Assembly/DOTween/Plugins/Core/PathCore/CatmullRomDecoder.cs:46)
DG.Tweening.Plugins.Core.PathCore.Path.GetPoint (Single perc, Boolean convertToConstantPerc) (at D:/DG/_Develop/__UNITY3_CLASSES/_Holoville/__DOTween/_DOTween.Assembly/DOTween/Plugins/Core/PathCore/Path.cs:100)
DG.Tweening.Plugins.Core.PathCore.Path.RefreshNonLinearDrawWps (DG.Tweening.Plugins.Core.PathCore.Path p) (at D:/DG/_Develop/__UNITY3_CLASSES/_Holoville/__DOTween/_DOTween.Assembly/DOTween/Plugins/Core/PathCore/Path.cs:178)
DG.Tweening.Plugins.Core.PathCore.Path.Draw (DG.Tweening.Plugins.Core.PathCore.Path p) (at D:/DG/_Develop/__UNITY3_CLASSES/_Holoville/__DOTween/_DOTween.Assembly/DOTween/Plugins/Core/PathCore/Path.cs:300)
DG.Tweening.Plugins.Core.PathCore.Path.Draw () (at D:/DG/_Develop/__UNITY3_CLASSES/_Holoville/__DOTween/_DOTween.Assembly/DOTween/Plugins/Core/PathCore/Path.cs:286)
DG.Tweening.Core.DOTweenComponent.OnDrawGizmos () (at D:/DG/_Develop/__UNITY3_CLASSES/_Holoville/__DOTween/_DOTween.Assembly/DOTween/Core/DOTweenComponent.cs:119)
UnityEngine.GUIUtility:ProcessEvent(Int32, IntPtr)



IndexOutOfRangeException: Array index is out of range.
DG.Tweening.Plugins.PathPlugin.SetChangeValue (DG.Tweening.Core.TweenerCore`3 t) (at D:/DG/_Develop/__UNITY3_CLASSES/_Holoville/__DOTween/_DOTween.Assembly/DOTween/Plugins/PathPlugin.cs:73)
DG.Tweening.Tweener.DoStartup[Vector3,Path,PathOptions] (DG.Tweening.Core.TweenerCore`3 t) (at D:/DG/_Develop/__UNITY3_CLASSES/_Holoville/__DOTween/_DOTween.Assembly/DOTween/Tweener.cs:147)
DG.Tweening.Core.TweenerCore`3[UnityEngine.Vector3,DG.Tweening.Plugins.Core.PathCore.Path,DG.Tweening.Plugins.Options.PathOptions].Startup () (at D:/DG/_Develop/__UNITY3_CLASSES/_Holoville/__DOTween/_DOTween.Assembly/DOTween/Core/TweenerCore.cs:168)
DG.Tweening.Tween.DoGoto (DG.Tweening.Tween t, Single toPosition, Int32 toCompletedLoops, UpdateMode updateMode) (at D:/DG/_Develop/__UNITY3_CLASSES/_Holoville/__DOTween/_DOTween.Assembly/DOTween/Tween.cs:194)
DG.Tweening.Core.TweenManager.Rewind (DG.Tweening.Tween t, Boolean includeDelay) (at D:/DG/_Develop/__UNITY3_CLASSES/_Holoville/__DOTween/_DOTween.Assembly/DOTween/Core/TweenManager.cs:752)
DG.Tweening.Core.TweenManager.Restart (DG.Tweening.Tween t, Boolean includeDelay, Single changeDelayTo) (at D:/DG/_Develop/__UNITY3_CLASSES/_Holoville/__DOTween/_DOTween.Assembly/DOTween/Core/TweenManager.cs:725)
DG.Tweening.TweenExtensions.Restart (DG.Tweening.Tween t, Boolean includeDelay, Single changeDelayTo) (at D:/DG/_Develop/__UNITY3_CLASSES/_Holoville/__DOTween/_DOTween.Assembly/DOTween/TweenExtensions.cs:204)
DG.Tweening.DOTweenPath.DORestart (Boolean fromHere) (at D:/DG/_Develop/__UNITY3_CLASSES/_Holoville/__DOTween/_DOTween.Assembly/DOTweenPro/DOTweenPath.cs:263)
TextFlyout.Update () (at Assets/Scripts/TextFlyout.cs:28)
`

**Comment by Demigiant** [comment]
Ahoy,

Sorry I was away and came back this evening. I'm going to check tomorrow (I made a few tests but I couldn't replicate the error, gonna try with the kill-inside-complete, though it's rather weird because I remember that was indeed a bug that I fixed a while ago :B). @Ladimaco can you confirm me that those are errors and not just warnings?

Cheers.

**Comment by Demigiant** [comment]
Ahoy,

I tried many ways to replicate this error, both in precedence with the creation of tweens inside OnComplete callbacks, and by killing tweens inside the OnComplete callback (both the same tween that owns the callback and other tweens), but everything goes smoothly here.

@Ladimaco can you show me the code that is creating the issue in your case?

**Comment by eintopf** [comment]
Did you check "So broken in 1.0.075 It works in 1.0.041 (Dotween 1.2.055)". Maybe you something in the commit history what changed in between what could cause these problems? 

Maybe another hint: For us it seems like it happens when we really played a lot of tweens. So more in the later phase of the game.

**Comment by Demigiant** [comment]
The only thing I changed which could vaguely influence that is that now, if a nested tween inside a Sequence fails, the Sequence cancels said tween and proceeds instead of canceling the whole Sequence. I made some tests with that too but it doesn't seem related. Though uhm, could you try [this version of DOTween](http://dotween.demigiant.com/tmp/DOTween_1_2_135b.zip)? It's the same exact version as the latest one, except I reverted that Sequence behaviour.

**Comment by eintopf** [comment]
Hm, is this compatible with DoTween Pro? If i add that to the project I get a ton of compile errors :/ related to TextMeshPro

**Comment by jakemhill** [comment]
Hey @Demigiant . Sorry to post on an old thread but I think it fits here, I seem to be having the exact same issue as the OP, using v1.2.335 + pro 1.0.187, looks like Android only. Getting thousands of hits with no useable stack trace other than what OP posted. 

Unclear if this was actually fixed, so not sure if this would count as a regression or just another instance of this bug. Could you please advise. 

Thanks

**Comment by Demigiant** [comment]
Ahoy @rebelincontrol. This has been fixed but there is one super rare occurrence that I never managed to replicate. But it's caused by using DOTween.Clear() (just don't know how), which is incorrect usage by itself (even if I'd love to be able to replicate this and prevent it anyway). Am I right in assuming you're using that? DOTween.Clear is kinda debug and should only be used when you want to remove DOTween completely and are not going to use it again in the same run. Otherwise you should use DOTween.Kill. Let me know :P

**Comment by jakemhill** [comment]
Ah sorry to disappoint but I just double checked and nowhere are we using DOTween.Clear()! I mostly use the Kill extension method throughout my code. 

Weird that it only seems to happen on Android? I've attached my DOTween settings, let me know if I can provide anything else to give an idea.. 

![image](https://user-images.githubusercontent.com/7241077/101237733-1a2ba780-36d3-11eb-9295-d2ec3a40be9e.png)

**Comment by Demigiant** [comment]
Ouch :O But the bug report is exactly like the OP, about IndexOutOfRangeException? Usually when a bug happens on a specific platform it's because of Unity's random initialization of classes, so that MonoBehaviourX's Awake/Start/Update/OnDestroy might be called before MonoBehaviourY's on one platform but after it on another.

**Comment by Pandanym** [comment]
Hi, I'm having the same issue with the same messages. It's only when not activating safe mode, but we need to see where errors are happening, so keeping safe mode on is not a solution for us.

We're using KillAll when switching scenes.
